File: src/AceRental.Application/Reservations/Dtos/ReservationDetailsDto.cs
using AceRental.Application.Clients.Dtos;
using AceRental.Application.Invoices.Dtos;
using AceRental.Domain.Enum;

namespace AceRental.Application.Reservations.Dtos
{
    public class ReservationDetailsDto
    {
        public Guid Id { get; set; }
        public required int ReservationNumber { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public required FinancialStatus FinancialStatus { get; set; }
        public required LogisticStatus LogisticStatus { get; set; }
        public required Workflow Workflow { get; set; }
        public decimal TotalHT { get; set; }
        public decimal TVA { get; set; } = 0.20m;
        public decimal TotalTTC => TotalHT * (1 + TVA);    
        public Guid ClientId { get; set; }
        public ClientReservationDto Client { get; set; } = null!;
        public int CurrentVersion { get; set; } = 1;
        public ICollection<ReservationItemDto> Items { get; set; } = [];
        public ICollection<InvoiceDto> Invoices { get; set; } = [];
    }
}

File: AceRental.FrontEnd/ClientApp/src/app/interfaces/reservation-details-dto.ts
import { FinancialStatus } from "../Enum/financial-status";
import { LogisticStatus } from "../Enum/logistic-status";
import { Workflow } from "../Enum/workflow";
import { ClientDto } from "./client-dto";
import { ReservationEquipmentsDto } from "./reservation-equipments-dto";
import { ReservationPacksDto } from "./reservation-packs-dto";
import { ReservationServicesDto } from "./reservation-services-dto";

export interface ReservationDetailsDto {
    Id?: string;
    ReservationNumber?: number;
    RaisonSociale?: string;
    StartDate: Date;
    EndDate: Date;
    FinancialStatus?: FinancialStatus;
    LogisticStatus?: LogisticStatus;
    Workflow: Workflow;
    TotalHT?: number;
    TVA?: number;
    TotalTTC?: number;
    ClientId: string;
    Client?: ClientDto;
    CurrentVersion?: number;
    Equipments: ReservationEquipmentsDto[];
    Packs: ReservationPacksDto[];
    Services: ReservationServicesDto[];
    // Invoices: List<InvoiceDto>;
    // Quotes: List<QuoteDto>;
    // Payments: List<PaymentDto>;

    CreatedAt?: Date;
    UpdatedAt?: Date;
}